includes/RestApi/FlowController.php

```diff
@@ -100,31 +97,32 @@ public function save_onboarding_flow_data( \WP_REST_Request $request ) {
 				array( 'status' => 404 )
 			);
 		}
-
-		if ( ! ( $flow_data = $this->read_details_from_wp_options() ) ) {
-			  $flow_data              = Flows::get_data();
-			  $flow_data['createdAt'] = time();
-			  // update default data if flow type is ecommerce
-			  $flow_data = $this->update_default_data_for_ecommerce( $flow_data );
-			  $this->save_details_to_wp_options( $flow_data );
+		
+		$flow_data = FlowService::read_flow_data_from_wp_option();
+		$updated_flow_data = array();
+		$flow_data = FlowService::update_post_call_data_recursive( $flow_data, $updated_flow_data, $params );
+		if(!is_array($flow_data)) {
+			return new \WP_Error(
+				'wrong_param_type_provided',
+				"Wrong Parameter Type Provided : ". $flow_data,
+				array( 'status' => 404 )
+			);
 		}
 
-		foreach ( $params as $key => $param ) {
-			if ( $value = $this->array_search_key( $key, $flow_data ) === false ) {
-				return new \WP_Error(
-					'wrong_param_provided',
-					"Wrong Parameter Provided : $key",
-					array( 'status' => 404 )
-				);
-			}
+		$default_flow_data = FlowService::get_default_flow_data();
+		$mismatch_key = FlowService::check_key_in_nested_array($params, $default_flow_data);
+		if ( ! empty($mismatch_key) )  {
+			return new \WP_Error(
+				'wrong_param_provided',
+				"Wrong Parameter Provided",
+				array( 'status' => 404 , 'Mismatched Parameter(s)' => $mismatch_key)
+			);
```

Comment:
Why are we searching flow data again, I thought lines 101-110 handled this?

Reply:
We are traversing through params in this case, here an error would be thrown if the user adds any key that does not exist in the blueprint, hence it needs to be handled separately

Reply:
This needs to go above the post recursive call then like we discussed

Reply:
Changes have been made